addOns/graphql/src/test/java/org/zaproxy/addon/graphql/GraphQlFingerprinterUnitTest.java

```diff
@@ -331,6 +338,72 @@ void shouldFingerprintValidData(String graphqlImpl, String response) throws Exce
         assertThat(discoveredEngine.get(0).getName(), is(equalTo(graphqlImpl)));
     }
 
+    @Test
+    void shouldFingerprintWithoutAddedHandler() throws Exception {
+        // Given
+        var url = UrlBuilder.build(endpointUrl);
+        nano.addHandler(new GraphQlResponseHandler(errorResponse("The query must be a string.")));
+        var fp = new GraphQlFingerprinter(url);
+        Logger logger = (Logger) LogManager.getLogger(GraphQlFingerprinter.class);
+        StringWriter writer = new StringWriter();
+        WriterAppender appender =
+                WriterAppender.newBuilder()
+                        .setName("TestAppender")
+                        .setLayout(
+                                PatternLayout.newBuilder()
+                                        .withPattern(PatternLayout.TTCC_CONVERSION_PATTERN)
+                                        .build())
+                        .setTarget(writer)
+                        .build();
+        appender.start();
+        try {
+            logger.addAppender(appender);
+            logger.setLevel(Level.ALL);
+            // When
+            fp.fingerprint();
+            // Then
+            assertNoNpe(writer.toString());
```

Comment:
IMO `assertNoErrors` is clearer, it can also check the alert was raised to make sure the expected code path was followed.

Reply:
Done

Reply:
There's already code asserting that the alert was raised, there's no reason to use spy.

Reply:
Tweaked